test/common/upstream/BUILD

```diff
@@ -142,9 +142,14 @@ envoy_benchmark_test(
     benchmark_binary = "eds_speed_test",
 )
 
-envoy_cc_test(
-    name = "health_checker_impl_test",
-    srcs = ["health_checker_impl_test.cc"],
+envoy_cc_test_library(
+    name = "health_checker_impl_test_lib",
+    srcs = [
+        "health_checker_impl_test.cc",
```

Comment:
suggest renaming to "health_checker_test_utils.h" and keeping the test as `health-checker_impl_test.cc`